src/en/07_ffi.md

```diff
@@ -130,7 +130,14 @@ The following types are considered C-compatible:
 - `repr(C)`-annotated `struct`,
 - `repr(C)` or `repr(Int)`-annotated `enum` with at least one variant and only
   fieldless variants (where `Int` is an integral primitive type),
-- pointers.
+- pointers,
+- an `Option<T>` where `T` is either
+  - `core::ptr::NonNull<U>` and `U` is a `Sized` C-compatible type, then it is
+      compatible to a `*const T` and `*mut T` pointer;
+  - `core::num::NonZero*`, then is compatible to the corresponding integral
+      primitive type;
+- a `repr(transparent)`-annotated `struct` with only one field, where that
```

Comment:
You're allowed to have more than one field if one is a align-1 ZST. This is common for PhantomData, for example.

Reply:
You're even allowed to have an arbitrary amount of align-1 ZSTs, however this seemed very similar to the ZST restrictions below. The list style itself is not exactly exhaustive in either case. I have not yet seen any such use whereas I did find use for all other cases that I've listed.